pkg/templates/hostmounts/template.go

```diff
@@ -0,0 +1,118 @@
+package hostmounts
+
+import (
+	"fmt"
+	"regexp"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/extract"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/hostmounts/internal/params"
+)
+
+const (
+	templateKey = "host-mounts"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Host Mounts",
+		Key:         templateKey,
+		Description: "Flag volume mounts of sensitive system directories",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(_ lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
```

Comment:
It would be cleaner to compile the regexes _outside_ the func, and then call matchString on the compiled regex instead of `p.Dirs`. This allows us to return an error at check instantiation time if one of the regexes is invalid. Also, we should probably return an error if `len(p.Dirs) == 0`?
```suggestion
			compiledRegexes := make([]*regexp.Regexp, 0, len(p.Dirs))
			for _, dir := range p.Dirs {
			    r, err := regexp.Compile(dir)
			    if err != nil {
			       return nil, errors.Wrapf(err, "invalid regex %s", dir)
			    }
			    compiledRegexes = append(compiledRegexes, r)
			}
			return func(_ lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
```

Reply:
good suggestion! done.